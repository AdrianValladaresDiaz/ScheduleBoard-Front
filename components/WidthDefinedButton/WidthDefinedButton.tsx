import React from "react";
import styled from "styled-components";

const GnarlyButton = styled.button`
  cursor: pointer;
  display: flex;
  background-color: transparent;
  border: none;
  align-items: center;
  & .content {
    font-size: ${(props) => props.theme.textSizeSmallText};
    line-height: ${(props) => props.theme.textSizeSmallText};
    width: min-content;
    margin-right: 5px;
    border-bottom: 2px solid black;
  }
  & .bigContent {
    font-size: 40px;
  }
`;

interface WidthDefinedButtonProps {
  onClickAction: Function;
  content: string | JSX.Element;
  bigContent: string;
  isDisabled?: boolean;
  title?: string;
}

const WidthDefinedButton = ({
  onClickAction,
  content,
  bigContent,
  isDisabled = false,
  title = "",
}: WidthDefinedButtonProps): JSX.Element => {
  return (
    <GnarlyButton
      title={title}
      disabled={isDisabled}
      onClick={(event: React.MouseEvent<HTMLButtonElement>) => onClickAction()}
    >
      <div className="content">{content}</div>
      <div className="bigContent">{bigContent}</div>
    </GnarlyButton>
  );
};

export default WidthDefinedButton;
