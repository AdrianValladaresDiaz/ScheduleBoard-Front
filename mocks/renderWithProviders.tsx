import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../redux/store";
import React, { ReactNode } from "react";

interface wrappedChildren {
  children: ReactNode;
}

const renderWithProviders = (component: JSX.Element) => {
  const Providers = ({ children }: wrappedChildren) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return render(component, { wrapper: Providers });
};

export default renderWithProviders;
