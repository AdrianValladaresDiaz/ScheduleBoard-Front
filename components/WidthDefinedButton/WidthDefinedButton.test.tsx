import { screen } from "@testing-library/react";
import renderWithProviders from "../../mocks/renderWithProviders";
import WidthDefinedButton from "./WidthDefinedButton";

describe("Given the WidthDefinedButton", () => {
  describe("When rendered", () => {
    test("It should match the snapshot", () => {
      renderWithProviders(
        <WidthDefinedButton
          bigContent="O"
          content="normal content"
          onClickAction={() => {}}
        />
      );

      const button = screen.getByRole("button");

      (expect(button) as any).toMatchSnapshot();
    });
  });
});
