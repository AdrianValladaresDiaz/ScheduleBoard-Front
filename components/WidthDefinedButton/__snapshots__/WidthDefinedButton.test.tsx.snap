// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the WidthDefinedButton When rendered It should match the snapshot 1`] = `
<button
  class="sc-bdvvtL hNfttX"
  title=""
>
  <div
    class="content"
  >
    normal content
  </div>
  <div
    class="bigContent"
  >
    O
  </div>
</button>
`;
